cleanDockerImages/cleanDockerImages.groovy

```diff
@@ -0,0 +1,299 @@
+
+    def RepoPath parentRepoPath = parentInfo.getRepoPath()
+    if(foundManifest){
+        if(listEndReached){
+            //reset Skip when the List ends
+            foundManifest=false
+        }
+        //Commenting Skipping logic
+        // log.debug "Skipping to dig folder " + parentRepoPath.getName()
+        //skipOtherDockerReposAfterManifestFound(parentRepoPath)
+        //return
+    }
+    List<ItemInfo> childItems = repositories.getChildren(parentRepoPath)
+
+
+    if (!childItems.isEmpty()) {
+
+        def listSize = childItems.size()
+        listEndReached=false
+        for (int i=0;i<listSize;i++){
+
+            if(i==(listSize-1)){
+                listEndReached = true
+            }
+            def ItemInfo childItem = childItems.get(i);
+            def RepoPath currentPath = childItem.getRepoPath()
+
+            if (childItem.isFolder()) {
+                simpleTraverse(childItem, revistFoldersList)
+
+            }
+            else{
+                log.debug "Scanning File: " + currentPath.getName()
+                if(isThisManifestFile(currentPath)){
+                    foundManifest=true
+                    //Get the properties here and delete based on policies
+                    //Implement DaysPassed Policy First and delete the images that qualify
+                    // And then aggregate the images info to group by image and sorted by create date for MaxCount policy
+                    if(checkDaysPassedForDelete(childItem)){
+                        log.debug("***** DELETE THE DOCKER IMAGE AS DAYS PASSED POLICY APPLIES " + parentRepoPath.getName())
+                        //Implement days passed policy and delete any docker images(tags) first
+                        deleteDockerImage(parentRepoPath)
+                    } else{
+
+                        def int maxCount = getMaxCountForDelete(childItem)
+                        // if maxCount property is zero then don't delete any!!
+                        if(maxCount >0){
+                            log.debug "Adding to IMAGES MAP :" + parentRepoPath
+                            def long parentCreatedDate = parentInfo.getCreated()
+                            def String parentId = parentRepoPath.getParent().getId()
+                            imagesCount.put(parentId, new Integer(maxCount))
+
+                            if(!imagesPathMap.containsKey(parentId)){
+                                log.debug "RESETTING ARRAY LIST :"
+                                revistFoldersList = new ArrayList()
+                                revistFoldersList.add(parentRepoPath)
+                                imagesPathMap.put(parentId,revistFoldersList)
+
+                            }else{
+                                revistFoldersList = imagesPathMap.get(parentId)
+                                revistFoldersList.add(parentRepoPath)
+                                imagesPathMap.put(parentId,revistFoldersList)
```

Comment:
@madhuredde there is no need to put the list in the map again since is was fetched from there.